hooks/useVideo.ts

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from "react";
+
+export const useVideo = (videoEl) => {
+  const [video, setVideo] = useState(null);
+  const [playing, setPlaying] = useState(false);
+
+  const toggle = () => setPlaying(!playing);
+
+  useEffect(() => {
+    playing ? videoEl.current?.play() : videoEl.current?.pause();
+  }, [playing]);
+
+  useEffect(() => {
+    videoEl?.current?.addEventListener("ended", () => setPlaying(false));
+    return () => {
+      videoEl?.current?.removeEventListener("ended", () => setPlaying(false));
+    };
+  }, [video]);
+
+  return [playing, toggle] as [boolean, () => void];
+};
```

Comment:
nit: If you have a spare moment, your IDE should really add a newline to the end of files. It messes with different UNIX/Terminal commands

https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file